### PR TITLE
Added an option to strip trailing whitespace on paste

### LINKED ISSUE #1888: Pasting command with a new-line will execute the command

### PR DESCRIPTION
**Pull request by steveniemitz**
Fixes #1888

### CODE REVIEW

**Comment by steveniemitz** [comment]
ping?

**Comment by ximion** [comment]
I see the sense in this patch - did you test the current PR?
D's string.stripRight() will indeed remove all whitespaces from the end of a line, but it will not remove any newline or tab characters, so since newlines aren't removed, this wouldn't fix #1888
I just looked over the changes quickly though, so maybe I missed something.

**Comment by steveniemitz** [comment]
> did you test the current PR

yeah, I've been running with this change since before I submitted the PR.

> D's string.stripRight() .. will not remove any newline or tab characters

This is false, the Dlang docs even include newline, linefeed, and tab characters as examples: 
https://dlang.org/library/std/string/strip_right.html

```
assert(stripRight("\n\t\v\rhello world\n\t\v\r") ==
       "\n\t\v\rhello world");

assert(stripRight([lineSep] ~ "hello world" ~ lineSep) ==
       [lineSep] ~ "hello world");
```

**Comment by ximion** [comment]
Right, I even looked at the page and completely misread it...

File: source/gx/tilix/prefeditor/prefdialog.d [hunk 1]
```diff
@@ -1577,6 +1577,11 @@ private:
         bh.bind(SETTINGS_STRIP_FIRST_COMMENT_CHAR_ON_PASTE_KEY, cbStrip, "active", GSettingsBindFlags.DEFAULT);
         add(cbStrip);
 
+        //Strip trailing whitespace on paste
+        CheckButton cbStripTrailing = new CheckButton(_("Strip trailing whitespace"));
```
Comment (ximion) [inline source/gx/tilix/prefeditor/prefdialog.d:1581]: Maybe name this "Strip trailing whitespaces and linebreak characters on paste", to make it a bit more clear *when* those whitespaces and linebreaks are removed, as well as to be similar in style to the preceding "Strip first character of paste if comment or variable declaration" option

Comment (steveniemitz) [reply source/gx/tilix/prefeditor/prefdialog.d:1581]: makes sense, I'll update the text

**Comment by ximion** [comment]
Looks good to me! Thank you for the patch!

**Comment by steveniemitz** [comment]
thank you!

**Pull request merged**